packages/core/src/tests/uma.test.ts

```diff
@@ -198,7 +200,81 @@ describe("uma", () => {
 
     const query = parseLnurlpRequest(queryUrl);
     expect(query.umaVersion).toBe(UmaProtocolVersion);
-    const verified = await verifyUmaLnurlpQuerySignature(query, publicKey);
+    const verified = await verifyUmaLnurlpQuerySignature(
+      query,
+      publicKey,
+      new InMemoryNonceValidator(oneWeekAgo),
+    );
+    expect(verified).toBe(true);
+  });
+
+  it("should throw for duplicate nonce when verifying signature", async () => {
+    const { privateKey, publicKey } = await generateKeypair();
+    const queryUrl = await getSignedLnurlpRequestUrl({
+      signingPrivateKey: privateKey,
+      receiverAddress: "$bob@vasp2.com",
+      senderVaspDomain: "vasp1.com",
+      isSubjectToTravelRule: true,
+    });
+
+    const query = parseLnurlpRequest(queryUrl);
+    const nonceCache = new InMemoryNonceValidator(1);
+    nonceCache.checkAndSaveNonce(query.nonce, 2);
+    try {
+      await verifyUmaLnurlpQuerySignature(query, publicKey, nonceCache);
+    } catch (e) {
+      if (!isError(e)) {
+        throw new Error("Invalid error type");
+      }
+      expect(e.message).toMatch(
+        /Invalid response nonce. Already seen this nonce or the timestamp is too old./,
+      );
+    }
```

Comment:
This pattern will test that *if* the verify call throws, it's the right error, but doesn't assert that it *does* throw. We should either:

- Add a failure after this try/catch and early return in the catch block, or
- Use `expect(...).toThrow()`

Same for all expected error cases in here.